src/boot/bootctl.c

```diff
@@ -1345,18 +1361,53 @@ static int verb_list(int argc, char *argv[], void *userdata) {
                 log_info("No boot loader entries found.");
         else {
                 size_t n;
+                if (arg_json_format != JSON_FORMAT_OFF) {
+                        //Format flags
+                        BootEntry *entry = NULL;
+                        _cleanup_(json_variant_unrefp) JsonVariant *output;
+                        char *options = NULL;
+                        char *initrd = NULL;
+
+                        printf("{");
```

Comment:
please do not generate json output manually, otherwise you won't get the various benefits the logic from json.c provides you with, i.e. pretty printing and so on.

generate the individual objects, then combine them into an array with json_variant_new_array()